[0.19] Add search field to community sidebar

Nutomic opened the pull request:
Add search bar to community sidebar as [discussed on Lemmy](https://lemmy.ml/post/32355919/19527965).

![Screenshot_20250701_162412-1](https://github.com/user-attachments/assets/f438e03d-eabd-4dfd-b15e-c6556a9ab57c)

dessalines reviewed (commented):
I don't think using form actions like that is going to work correctly for the SPA (IE not doing any page reloading).

There is an example of searching in `communities.tsx`. Here are the main places to look at:

- https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/components/community/communities.tsx#L244
- https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/components/community/communities.tsx#L262
- https://github.com/LemmyNet/lemmy-ui/blob/main/src/shared/components/community/communities.tsx#L320

You might want to also rebase this off of `release/v0.19` first, to be able to test it. Then do another PR to main after that ones merged.

Nutomic commented:
I did test with 0.19 but opened the PR for main as there were no conflicts anyway. Changed the target branch now so its easier to test. Also changed it to use linkEvent like the example you linked.

SleeplessOne1917 commented on src/shared/components/community/sidebar.tsx:
```diff
@@ -162,6 +164,27 @@ export class Sidebar extends Component<SidebarProps, SidebarState> {
+                  >
+                    <input
+                      name="q"
+                      type="text"
```
Could make this `type="search"` instead.

Nutomic commented on src/shared/components/community/sidebar.tsx:
```diff
@@ -162,6 +164,27 @@ export class Sidebar extends Component<SidebarProps, SidebarState> {
                   />
                   {this.canPost && this.createPost()}
                   {myUserInfo && this.blockCommunity()}
+                  <form
+                    class="d-flex"
+                    onSubmit={linkEvent(this, this.handleSearchSubmit)}
```
It seems `<search>` element is not supported: 
```
rc/shared/components/community/sidebar.tsx:167:19 - error TS2339: Property 'search' does not exist on type 'JSX.IntrinsicElements'.

167                   <search>
```
